File: library/src/test/java/com/mindera/skeletoid/logs/utils/LogAppenderUtilsUnitTests.java
package com.mindera.skeletoid.logs.utils;

import com.mindera.skeletoid.generic.AndroidUtils;

import org.junit.Test;

import static junit.framework.Assert.assertEquals;

public class LogAppenderUtilsUnitTests {

    private String mPackageName = "my.package.name";

    @Test(expected = UnsupportedOperationException.class)
    public void testConstructor() {
        new LogAppenderUtils();
    }

    @Test
    public void testGetLogString() {
        assertEquals("A B C ", LogAppenderUtils.getLogString("A", "B", "C"));
    }

    @Test
    public void testGetLogStringNull() {
        assertEquals("", LogAppenderUtils.getLogString(new String[0]));
    }

    @Test
    public void testGetMethodName() {
        assertEquals("testGetMethodName", LogAppenderUtils.getMethodName(LogAppenderUtilsUnitTests.class));
    }

    @Test
    public void testGetMethodNameInvalid() {
        assertEquals("UnknownMethod", LogAppenderUtils.getMethodName(AndroidUtils.class));
    }

    @Test
    public void testGetTag() {
        assertEquals(LogAppenderUtilsUnitTests.class.getCanonicalName(), LogAppenderUtils.getTag(LogAppenderUtilsUnitTests.class, false, mPackageName, false));
    }

    @Test
    public void testGetTagWithPackageName() {
        assertEquals(mPackageName + "/" + LogAppenderUtilsUnitTests.class.getCanonicalName(), LogAppenderUtils.getTag(LogAppenderUtilsUnitTests.class, true, mPackageName, false));
    }

    @Test
    public void testGetTagWithPackageNameAndMethodName() {
        assertEquals(mPackageName + "/" + LogAppenderUtilsUnitTests.class.getCanonicalName() + ".testGetTagWithPackageNameAndMethodName",
                LogAppenderUtils.getTag(LogAppenderUtilsUnitTests.class, true, mPackageName, true));
    }

    @Test
    public void testGetTagWithMethodName() {
        assertEquals(LogAppenderUtilsUnitTests.class.getCanonicalName() + ".testGetTagWithMethodName",
                LogAppenderUtils.getTag(LogAppenderUtilsUnitTests.class, false, mPackageName, true));
    }

    @Test
    public void testGetObjectHash() {
        assertEquals("[OID#" + this.hashCode() + "] ", LogAppenderUtils.getObjectHash(this));
    }

    @Test
    public void testGetObjectHashNull() {
        assertEquals("[!!!NULL INSTANCE!!!] ", LogAppenderUtils.getObjectHash(null));
    }


}

File: base/src/test/java/com/mindera/skeletoid/logs/utils/LogAppenderUtilsUnitTests.kt
package com.mindera.skeletoid.logs.utils

import com.mindera.skeletoid.generic.AndroidUtils
import com.mindera.skeletoid.logs.utils.LogAppenderUtils.getLogString
import com.mindera.skeletoid.logs.utils.LogAppenderUtils.getMethodName
import com.mindera.skeletoid.logs.utils.LogAppenderUtils.getObjectHash
import com.mindera.skeletoid.logs.utils.LogAppenderUtils.getTag
import org.junit.Assert
import org.junit.Test

class LogAppenderUtilsUnitTests {
    companion object{
        private const val packageName = "my.package.name"
    }

    @Test
    fun testGetLogString() {
        Assert.assertEquals("A B C ", getLogString("A", "B", "C"))
    }

    @Test
    fun testGetLogStringEmptyString() {
        Assert.assertEquals("", getLogString())
    }

    @Test
    fun testGetMethodName() {
        Assert.assertEquals(
            "testGetMethodName", getMethodName(
                LogAppenderUtilsUnitTests::class.java
            )
        )
    }

    @Test
    fun testGetMethodNameInvalid() {
        Assert.assertEquals(
            "UnknownMethod", getMethodName(
                AndroidUtils::class.java
            )
        )
    }

    @Test
    fun testGetTag() {
        Assert.assertEquals(
            LogAppenderUtilsUnitTests::class.java.canonicalName, getTag(
                LogAppenderUtilsUnitTests::class.java, false, packageName, false
            )
        )
    }

    @Test
    fun testGetTagWithPackageName() {
        Assert.assertEquals(
            packageName + "/" + LogAppenderUtilsUnitTests::class.java.canonicalName,
            getTag(
                LogAppenderUtilsUnitTests::class.java, true, packageName, false
            )
        )
    }

    @Test
    fun testGetTagWithPackageNameAndMethodName() {
        Assert.assertEquals(
            packageName + "/" + LogAppenderUtilsUnitTests::class.java.canonicalName + ".testGetTagWithPackageNameAndMethodName",
            getTag(LogAppenderUtilsUnitTests::class.java, true, packageName, true)
        )
    }

    @Test
    fun testGetTagWithMethodName() {
        Assert.assertEquals(
            LogAppenderUtilsUnitTests::class.java.canonicalName + ".testGetTagWithMethodName",
            getTag(
                LogAppenderUtilsUnitTests::class.java,
                false,
                packageName,
                true
            )
        )
    }

    @Test
    fun testGetObjectHash() {
        Assert.assertEquals(
            "[LogAppenderUtilsUnitTests#" + this.hashCode() + "] ",
            getObjectHash(this)
        )
    }
}